docs/schema-definition.md

```diff
@@ -105,63 +105,80 @@ $schema = new Schema($config);
 
 ## Lazy loading of types
 
-By default, the schema will scan all of your type, field and argument definitions to serve GraphQL queries.
-It may cause performance overhead when there are many types in the schema.
+If your schema makes use of a large number of complex or dynamically-generated types, they can become a performance concern. There are a few best practices that can lessen their impact:
 
-In this case, it is recommended to pass the **typeLoader** option to the schema constructor and define all
-of your object **fields** as callbacks.
+1. Use a type loader. This will put you in a position to implement your own caching and lookup strategies, and GraphQL won't need to preload a map of all known types to do its work.
 
-Type loading is very similar to PHP class loading, but keep in mind that the **typeLoader** must
-always return the same instance of a type. A good way to ensure this is to use a type registry:
```

Comment:
This constraint is important and no longer mentioned

Reply:
Moved to a comment:
39f35486

Good enough?